core/src/storage/traits/pull.rs

```diff
@@ -0,0 +1,274 @@
+                )
+            }
+        }
+
+        impl<$($frag),*> PullAt for ($($frag),* ,)
+        where
+            Self: scale::Decode,
+        {
+            fn pull_at(at: Key) -> Self {
+                pull_single_cell::<Self>(at)
+            }
+        }
+    }
+}
+impl_pull_tuple!(A);
+impl_pull_tuple!(A, B);
+impl_pull_tuple!(A, B, C);
+impl_pull_tuple!(A, B, C, D);
+impl_pull_tuple!(A, B, C, D, E);
+impl_pull_tuple!(A, B, C, D, E, F);
+impl_pull_tuple!(A, B, C, D, E, F, G);
+impl_pull_tuple!(A, B, C, D, E, F, G, H);
+impl_pull_tuple!(A, B, C, D, E, F, G, H, I);
+impl_pull_tuple!(A, B, C, D, E, F, G, H, I, J);
+
+impl PullForward for () {
+    fn pull_forward(_ptr: &mut KeyPtr) -> Self {}
+}
+
+impl PullAt for () {
+    fn pull_at(_at: Key) -> Self {}
+}
+
+impl<T> PullForward for PhantomData<T> {
+    fn pull_forward(_ptr: &mut KeyPtr) -> Self {
+        Default::default()
+    }
+}
+
+impl<T> PullAt for PhantomData<T> {
+    fn pull_at(_at: Key) -> Self {
+        Default::default()
+    }
+}
+
+impl<T> PullForward for Option<T>
+where
+    T: PullForward + StorageSize,
+{
+    fn pull_forward(ptr: &mut KeyPtr) -> Self {
+        // We decode as `()` because at this point we are not interested
+        // in the actual value, we just want to know if there exists a value
+        // at all.
+        match env::get_contract_storage::<()>(ptr.current()) {
+            Some(_) => Some(<T as PullForward>::pull_forward(ptr)),
+            None => None,
+        }
+    }
+}
+
+impl<T> PullAt for Option<T>
+where
+    Self: scale::Decode,
+{
+    fn pull_at(at: Key) -> Self {
+        pull_single_cell::<Self>(at)
+    }
+}
+
+impl<T, E> PullForward for Result<T, E>
+where
+    T: PullForward,
+    E: PullForward,
+{
+    fn pull_forward(ptr: &mut KeyPtr) -> Self {
+        match pull_single_cell::<u8>(ptr.next_for::<u8>()) {
+            0 => Ok(<T as PullForward>::pull_forward(ptr)),
+            1 => Err(<E as PullForward>::pull_forward(ptr)),
+            _ => unreachable!("found invalid Result discriminator"),
+        }
+    }
+}
+
+impl<T, E> PullAt for Result<T, E>
+where
+    Self: scale::Decode,
+{
+    fn pull_at(at: Key) -> Self {
+        pull_single_cell::<Self>(at)
+    }
+}
+
+impl PullForward for ink_prelude::string::String {
+    fn pull_forward(ptr: &mut KeyPtr) -> Self {
+        <Self as PullAt>::pull_at(ptr.next_for::<Self>())
+    }
+}
+
+impl PullAt for ink_prelude::string::String {
+    fn pull_at(at: Key) -> Self {
+        pull_single_cell::<Self>(at)
+    }
+}
+
+impl<T> PullForward for ink_prelude::boxed::Box<T>
+where
+    T: PullForward,
+{
+    fn pull_forward(ptr: &mut KeyPtr) -> Self {
+        Self::new(<T as PullForward>::pull_forward(ptr))
+    }
+}
+
+impl<T> PullAt for ink_prelude::boxed::Box<T>
+where
+    T: PullAt,
+{
+    fn pull_at(at: Key) -> Self {
+        Self::new(<T as PullAt>::pull_at(at))
+    }
+}
+
+const _: () = {
+    use ink_prelude::collections::{
+        BTreeSet,
```

Comment:
I can't seem to find the `BTreeSet` module? Did you commit it?

Reply:
This is the Rust `alloc` crate `BTreeSet` not ink!'s `storage::BTreeMap`